From 17f98e374b3769864c80b9ee6334ec195ff80f9c Mon Sep 17 00:00:00 2001
From: Cici Ruan <cuicuiruan@google.com>
Date: Wed, 24 Jan 2024 10:37:00 -0800
Subject: [PATCH] Revert "Mark LOG(FATAL) [[noreturn]]"

This reverts commit 42e36c6299f74529a140c5f3250afcaa83514cc3.
This reverts commit cd0e77341ce8f050f0e3d44a417b12ad83a561d0.

Change-Id: I0a9629caac3d992f89875ec4326f0c42c036a618
---
 base/logging.h                 | 38 ++++++++++++++------------------
 base/logging_nocompile.nc      | 40 ----------------------------------
 mojo/core/node_channel.cc      |  1 +
 mojo/public/c/system/thunks.cc |  1 +
 4 files changed, 18 insertions(+), 62 deletions(-)
 delete mode 100644 base/logging_nocompile.nc

diff --git a/base/logging.h b/base/logging.h
index ae1bbeef9a..23e112d1fc 100644
--- a/base/logging.h
+++ b/base/logging.h
@@ -416,11 +416,10 @@ constexpr LogSeverity LOGGING_0 = LOGGING_ERROR;
 // As special cases, we can assume that LOG_IS_ON(FATAL) always holds. Also,
 // LOG_IS_ON(DFATAL) always holds in debug mode. In particular, CHECK()s will
 // always fire if they fail.
-// FATAL is always enabled and required to be resolved in compile time for
-// LOG(FATAL) to be properly understood as [[noreturn]].
-#define LOG_IS_ON(severity)                                     \
-  (::logging::LOGGING_##severity == ::logging::LOGGING_FATAL || \
-   ::logging::ShouldCreateLogMessage(::logging::LOGGING_##severity))
+// TODO(crbug.com/1409729): Make sure LOG(FATAL) is understood as [[noreturn]]
+// by making sure the compiler knowns that it's unconditionally ON.
+#define LOG_IS_ON(severity) \
+  (::logging::ShouldCreateLogMessage(::logging::LOGGING_##severity))
 
 // Define a default ENABLED_VLOG_LEVEL if it is not defined. The macros allows
 // code to enable vlog level at build time without the need of --vmodule
@@ -530,22 +529,12 @@ BASE_EXPORT extern std::ostream* g_swallow_stream;
 
 #if DCHECK_IS_ON()
 
-// All of these definitions use DLOG_IS_ON() rather than define to their LOG()
-// equivalents, as DLOG(FATAL) and friends can't be understood as [[noreturn]]
-// but LOG(FATAL) is.
-#define DLOG_IS_ON(severity) \
-  (::logging::ShouldCreateLogMessage(::logging::LOGGING_##severity))
-
-#define DLOG(severity) LAZY_STREAM(LOG_STREAM(severity), DLOG_IS_ON(severity))
-#define DLOG_IF(severity, condition) \
-  LAZY_STREAM(LOG_STREAM(severity), DLOG_IS_ON(severity) && (condition))
-#define DPLOG(severity) LAZY_STREAM(PLOG_STREAM(severity), DLOG_IS_ON(severity))
-#define DPLOG_IF(severity, condition) \
-  LAZY_STREAM(PLOG_STREAM(severity), DLOG_IS_ON(severity) && (condition))
+#define DLOG_IS_ON(severity) LOG_IS_ON(severity)
+#define DLOG_IF(severity, condition) LOG_IF(severity, condition)
+#define DLOG_ASSERT(condition) LOG_ASSERT(condition)
+#define DPLOG_IF(severity, condition) PLOG_IF(severity, condition)
 #define DVLOG_IF(verboselevel, condition) VLOG_IF(verboselevel, condition)
 #define DVPLOG_IF(verboselevel, condition) VPLOG_IF(verboselevel, condition)
-#define DLOG_ASSERT(condition) \
-  DLOG_IF(FATAL, !(condition)) << "Assert failed: " #condition ". "
 
 #else  // DCHECK_IS_ON()
 
@@ -554,17 +543,22 @@ BASE_EXPORT extern std::ostream* g_swallow_stream;
 // Contrast this with DCHECK et al., which has different behavior.
 
 #define DLOG_IS_ON(severity) false
-#define DLOG(severity) EAT_STREAM_PARAMETERS
 #define DLOG_IF(severity, condition) EAT_STREAM_PARAMETERS
-#define DPLOG(severity) EAT_STREAM_PARAMETERS
+#define DLOG_ASSERT(condition) EAT_STREAM_PARAMETERS
 #define DPLOG_IF(severity, condition) EAT_STREAM_PARAMETERS
 #define DVLOG_IF(verboselevel, condition) EAT_STREAM_PARAMETERS
 #define DVPLOG_IF(verboselevel, condition) EAT_STREAM_PARAMETERS
-#define DLOG_ASSERT(condition) EAT_STREAM_PARAMETERS
 
 #endif  // DCHECK_IS_ON()
 
+#define DLOG(severity)                                          \
+  LAZY_STREAM(LOG_STREAM(severity), DLOG_IS_ON(severity))
+
+#define DPLOG(severity)                                         \
+  LAZY_STREAM(PLOG_STREAM(severity), DLOG_IS_ON(severity))
+
 #define DVLOG(verboselevel) DVLOG_IF(verboselevel, true)
+
 #define DVPLOG(verboselevel) DVPLOG_IF(verboselevel, true)
 
 // Definitions for DCHECK et al.
diff --git a/base/logging_nocompile.nc b/base/logging_nocompile.nc
deleted file mode 100644
index fbf6c2e8cf..0000000000
--- a/base/logging_nocompile.nc
+++ /dev/null
@@ -1,40 +0,0 @@
-// Copyright 2023 The Chromium Authors
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-// This is a "No Compile Test" suite.
-// https://dev.chromium.org/developers/testing/no-compile-tests
-
-#include "base/logging.h"
-
-// LOG(FATAL) must be understood as [[noreturn]].
-int Foo() {
-  LOG(FATAL) << "I am [[noreturn]]!";
-  return 42;  // expected-error {{'return' will never be executed}}
-}
-
-int Foo2() {
-  LOG_ASSERT(false) << "I am [[noreturn]] (sometimes)!";
-  return 42;  // expected-error {{'return' will never be executed}}
-}
-
-// It's important that our logging macros agree on [[noreturn]] in all build
-// configurations (or dead-code warnings become impossible to satisfy). As such
-// neither LOG(DFATAL) or DLOG(FATAL) may be understood as [[noreturn]]. This
-// non-void function not returning a value after LOG(DFATAL) and DLOG(FATAL)
-// should always be a compile error due to a missing return statement.
-int Bar() {
-  // No LOG(DFATAL) macros should be understood as [[noreturn]] under any build
-  // configurations.
-  LOG(DFATAL) << "I am not [[noreturn]]!";
-  LOG_IF(DFATAL, true) << "I am not [[noreturn]]!";
-  PLOG(DFATAL) << "I am not [[noreturn]]!";
-  PLOG_IF(DFATAL, true) << "I am not [[noreturn]]!";
-
-  // Same as above but DLOG(FATAL) instead of LOG(DFATAL).
-  DLOG(FATAL) << "I am not [[noreturn]]!";
-  DLOG_IF(FATAL, true) << "I am not [[noreturn]]!";
-  DPLOG(FATAL) << "I am not [[noreturn]]!";
-  DPLOG_IF(FATAL, true) << "I am not [[noreturn]]!";
-  DLOG_ASSERT(false) << "I am not [[noreturn]]!";
-}  // expected-error {{non-void function does not return a value}}
diff --git a/mojo/core/node_channel.cc b/mojo/core/node_channel.cc
index afd65da19e..bdf06bf7c5 100644
--- a/mojo/core/node_channel.cc
+++ b/mojo/core/node_channel.cc
@@ -255,6 +255,7 @@ scoped_refptr<NodeChannel> NodeChannel::Create(
     const ProcessErrorCallback& process_error_callback) {
 #if BUILDFLAG(IS_NACL)
   LOG(FATAL) << "Multi-process not yet supported on NaCl-SFI";
+  return nullptr;
 #else
   return new NodeChannel(delegate, std::move(connection_params),
                          channel_handle_policy, io_task_runner,
diff --git a/mojo/public/c/system/thunks.cc b/mojo/public/c/system/thunks.cc
index 4467e130e3..bf6c3a46dd 100644
--- a/mojo/public/c/system/thunks.cc
+++ b/mojo/public/c/system/thunks.cc
@@ -49,6 +49,7 @@ MojoResult NotImplemented(const char* name) {
       << "Mojo has not been initialized in this process. You must call "
       << "either mojo::core::Init() as an embedder, or |MojoInitialize()| if "
       << "using the mojo_core shared library.";
+  return MOJO_RESULT_UNIMPLEMENTED;
 }
 
 }  // namespace
-- 
2.43.0.429.g432eaa2c6b-goog

